src/components/templates/ListView/index.tsx

```diff
@@ -59,14 +59,17 @@ const filterByColumns: Array<any> = [];
 
 export default function ListView(props) {
   const { getPConnect, bInForm } = props;
-  const { globalSearch, presets, referenceList, rowClickAction, selectionMode, referenceType, payload, parameters } = props;
+  const { globalSearch, presets, referenceList, rowClickAction, selectionMode, payload, parameters } = props;
 
   const thePConn = getPConnect();
   const componentConfig = thePConn.getComponentConfig();
   const resolvedConfigProps = thePConn.getConfigProps();
 
-  /** pyGUID is used for Data classes and pyID is for Work classes */
-  const rowID = referenceType === 'Case' ? 'pyID' : 'pyGUID';
+  /** By default, pyGUID is used for Data classes and pyID is for Work classes as row-id/key */
+  // const rowID = referenceType === 'Case' ? 'pyID' : 'pyGUID';
+
+  /** Making rowID to be dynamic and not to be restricted to pyGUID or pyID */
+  const rowID = props.compositeKeys[0];
```

Comment:
I suspect it should be something more like:
const rowID = props.compositeKeys && props.compositeKeys[0] ? props.compositeKeys[0] : (referenceType === 'Case' ? 'pyID' : 'pyGUID');

Reply:
Yes, I did something like this before only assuming we would always get compositeKeys which isn't the case. We should have a test for this as I didn't catch this even after running all the tests multiple times. Will add a story for this in the backlog.